drivers/platform/pico/pico_spi.c

```diff
@@ -125,27 +100,70 @@ int32_t pico_spi_init(struct no_os_spi_desc **desc,
 		cpha = SPI_CPHA_1;
 		break;
 	default:
-		ret = -EINVAL;
-		goto error;
-		break;
+		return -EINVAL;
 	}
 
 	spi_set_format(pico_spi->spi_instance, data_bits, cpol, cpha, SPI_MSB_FIRST);
+
+	/* Set SPI pins */
+	gpio_set_function(pico_spi->spi_rx_pin, GPIO_FUNC_SPI);
+	gpio_set_function(pico_spi->spi_sck_pin, GPIO_FUNC_SPI);
+	gpio_set_function(pico_spi->spi_tx_pin, GPIO_FUNC_SPI);
+	gpio_set_function(pico_spi->spi_cs_pin, GPIO_FUNC_SPI);
+
+	/* Enable SPI */
+	hw_set_bits(&spi_get_hw(pico_spi->spi_instance)->cr1, SPI_SSPCR1_SSE_BITS);
+
+	return 0;
+}
+
+/**
+ * @brief Initialize the SPI communication peripheral.
+ * @param desc  - The SPI descriptor.
+ * @param param - The structure that contains the SPI parameters.
+ * @return 0 in case of success, error code otherwise.
+ */
+int32_t pico_spi_init(struct no_os_spi_desc **desc,
+		      const struct no_os_spi_init_param *param)
+{
+	int32_t ret;
+	struct no_os_spi_desc *descriptor;
+	struct pico_spi_desc *pico_spi;
+	struct pico_spi_init_param *pico_spi_ip;
+
+	if (!desc || !param || !param->extra)
+		return -EINVAL;
+
+	descriptor = (struct no_os_spi_desc *)calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	pico_spi = (struct pico_spi_desc *)calloc(1, sizeof(*pico_spi));
+	if (!pico_spi) {
+		ret = -ENOMEM;
+		goto error;
```

Comment:
should we free `pico_spi` in this case?

Reply:
freeing a null pointer will not generate an error, however I added a new label to free only the descriptor in this case.